props/saber_fett263_buttons.h

```diff
@@ -293,248 +664,3398 @@ class SaberFett263Buttons : public virtual PropBase {
+        if (a > twist_menu_ * 2/3) {
+          current_menu_angle_ += twist_menu_;
+          if (current_menu_angle_ > M_PI) current_menu_angle_ -= M_PI * 2;
+          Event(BUTTON_NONE, EVENT_TWIST_RIGHT);
+          last_rotate_millis_ = millis();
+        }
+        if (a < -twist_menu_ * 2/3) {
+          current_menu_angle_ -= twist_menu_;
+          if (current_menu_angle_ < M_PI) current_menu_angle_ += M_PI * 2;
+          Event(BUTTON_NONE, EVENT_TWIST_LEFT);
+          last_rotate_millis_ = millis();
+        }
+      }
+    }
+  }
+
+#ifdef FETT263_EDIT_MODE_MENU
+#define SUBMENUS 9
+#define COLOR_SUB 15
+#define COLOR_OPTIONS 3
+#define GESTURE_OPTIONS 12
+#define SET_SUBMENUS 6
+#define STYLE_SETTINGS 10
+#endif
+
+  enum MenuType {
+    MENU_TOP,
+    MENU_PRESET,
+    MENU_VOLUME,
+    TRACK_PLAYER,    
+    MENU_REHEARSE,
+#ifdef FETT263_EDIT_MODE_MENU
+    MENU_STYLE,
+    MENU_COLOR,
+    MENU_COLOR_CHANGE,
+    MENU_FONT,
+    MENU_TRACK,
+    MENU_LENGTH,
+    MENU_COPY,
+    MENU_DELETE,
+    MENU_RESET,
+    MENU_EXIT,
+    MENU_COLOR_SUB,
+    MENU_COLOR_MODE,
+    MENU_EFFECT,
+    MENU_RESET_COLOR,
+#if NUM_BLADES > 1
+    MENU_COPY_COLOR,
+    MENU_BLADE_STYLE,
+    MENU_BLADE_COLOR,
+    MENU_BLADE_COPY,
+    MENU_BLADE_SETTING,
+    MENU_BLADE_LENGTH,
+#endif
+    MENU_GESTURE_SUB,
+    MENU_SWINGON,
+    MENU_TWISTON,
+    MENU_THRUSTON,
+    MENU_STABON,
+    MENU_FORCEPUSH,
+    MENU_TWISTOFF,
+    MENU_POWERLOCK,
+    MENU_SETTING_SUB,
+    MENU_STYLE_SETTING_SUB,
+    MENU_STYLE_OPTION,
+    MENU_IGNITION_OPTION,
+    MENU_IGNITION_TIME,
+    MENU_RETRACTION_OPTION,
+    MENU_RETRACTION_TIME,
+    MENU_LOCKUP_POSITION,
+    MENU_DRAG_SIZE,
+    MENU_EMITTER_SIZE,
+    MENU_PREON_OPTION,
+    MENU_PREON_SIZE,
+    MENU_DIM_BLADE,
+    MENU_CLASH_THRESHOLD,
+    MENU_SWINGON_SPEED,
+    MENU_FORCEPUSH_LENGTH,
+    MENU_LOCKUP_DELAY,
+    MENU_CLASH_DETECT,
+    MENU_MAX_CLASH,
+    COLOR_BASE,
+    COLOR_ALT,
+    COLOR_BLAST,
+    COLOR_CLASH,
+    COLOR_LOCKUP,
+    COLOR_DRAG,
+    COLOR_LB,
+    COLOR_STAB,
+    COLOR_PREON,
+    COLOR_IGNITE,
+    COLOR_RETRACT,
+    COLOR_PSTOFF,
+    COLOR_SWING,
+    COLOR_EMITTER,
+    COLOR_OFF,
+#endif  
+  };
+
+  enum MainMenu {
+    EDIT_STYLE = 1,
+    EDIT_COLORS = 2,
+    EDIT_FONT = 3,
+    EDIT_TRACK = 4,
+    EDIT_SETTINGS = 5,
+    COPY_PRESET = 6,
+    DELETE_PRESET = 7,
+    EXIT_MENU = 8,
+  };
+  
+  enum ColorSubMenu {
+    EDIT_EFFECT_COLOR = 1,
+#if NUM_BLADES == 1
+    RESET_COLORS = 2,
+#else
+    COPY_COLORS = 2,
+    RESET_COLORS = 3,
+#endif
+  };
+  enum ColorOptions {
+    COLOR_LIST_MODE = 0,
+    COLOR_HUE_MODE = 1,
+    WHITE_MODE = 2,
+    BLACK_MODE = 3,
+  }; 
+   
+  enum SettingsMenu {
+    EDIT_VOLUME = 1,
+    EDIT_STYLE_SETTINGS = 2,
+    EDIT_CONTROL_SETTINGS = 3,
+    EDIT_CLASH_THRESHOLD = 4,
+    EDIT_BLADE_LENGTH = 5,
+    EDIT_BRIGHTNESS = 6,
+  };
+   
+  enum GestureControls {
+    SWINGON_IGNITION = 1,
```

Comment:
Because C++ is kind of dumb, it's very helpful to have all your enum values have common prefixes (or postfixes), like:
GESTURE_SWINGON_SPEED
GESTURE_TWIST_IGNITION
GESTURE_STAB_IGNITION

etc.

Reply:
updated